rpc/proto/mayastor.proto

```diff
@@ -258,6 +260,23 @@ message ListNexusReply {
   repeated Nexus nexus_list = 1;
 }
 
+// represents a nexus device
+message NexusV2 {
+  string name = 1;             // name of the nexus
+  string uuid = 2;             // UUID of the bdev
+  uint64 size = 3;             // size of the volume in bytes
+  NexusState state = 4;        // current state of the nexus
```

Comment:
Now that we have a NexusV2 (so MOAC won't be affected), could we return the actual nexus and children states and not the "simplified" states that MOAC wanted?

Reply:
From what I can tell, it's only the `ChildState` that's mapped into fewer `rpc::ChildState`s, do you mean you want `ChildState`s return directly (I assume the reason is dropped from the `Faulted` state)?

Reply:
The NexusState is also mapped IIRC, there's no Init or Closed in the protobuf.
Yes I think at some point we discussed getting the states returned directly @gila @paulyoong  ?
Yeah the reason is dropped from the Faulted state, I'm actually not sure how we'd do that in protobuf :/

Reply:
In the case of `NexusState` the mapping is done in `nexus_bdev` itself, not just in the RPC layer. This is looking to be more than a quick bug fix, please raise a separate ticket with some requirements.

Reply:
We have this old one, I guess it fits: https://mayadata.atlassian.net/browse/CAS-608